[PM-24289] fix popout windows on wayland (browser)

krjan02 opened the pull request:
## 🎟️ Tracking

https://github.com/bitwarden/clients/issues/9045
https://github.com/bitwarden/clients/issues/18743

## 📔 Objective

Bitwarden's popup windows in the browser extension currently rely on precise `left` and `top` coordinates derived from the main browser window's position. While effective on X11, these coordinates are consistently reported as `0` by Chrome on Wayland, due to Wayland's design philosophy of abstracting global window positions from client applications.

This meant our popups were always appearing in the top-left corner of the screen on Wayland, instead of their intended relative position.

This commit introduces a check: if `senderWindow.left` and `senderWindow.top` are both `0` (indicating a Wayland-like environment), we explicitly set the new popup's `left` and `top` properties to 0. This allows the Wayland compositor to use its default placement logic for the window, ensuring it opens reliably and in a sensible location, rather than attempting to apply invalid coordinates.

For X11 and other environments where coordinates are valid, the original, precise positioning logic remains unchanged. This ensures consistent and functional popup behavior across different Linux display servers.

## 📸 Screenshots

<img width="1588" height="1131" alt="image" src="https://github.com/user-attachments/assets/864466e4-8025-47c6-babf-4888c3fc8f68" />

## ⏰ Reminders before review

- Contributor guidelines followed ✅
- All formatters and local linters executed and passed ✅
- Written new unit and / or integration tests where applicable - Not applicable
- Protected functional changes with optionality (feature flags)  - Not applicable
- Used internationalization (i18n) for all UI strings  - Not applicable
- CI builds passed ✅
- Communicated to DevOps any deployment requirements 
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

krjan02 committed:
fix popout windows on wayland (browser)

krjan02 committed:
Merge branch 'bitwarden:main' into main

Alexandero89 commented on apps/browser/src/platform/browser/browser-popup-utils.ts:
```diff
@@ -145,17 +145,33 @@ class BrowserPopupUtils {
-      ...windowOptions,
-      url: BrowserPopupUtils.buildPopoutUrl(extensionUrlPath, singleActionKey),
-    };
+    const waylandLike = senderWindow.left === 0 && senderWindow.top === 0;
```
@krjan02 sorry to pressure you, but could you maybe add the comments if you find some time?
I think the pr looks pretty good and it seems this is the final change needed so it will get merged.
Thanks!

lucemans commented on apps/browser/src/platform/browser/browser-popup-utils.ts:
```diff
@@ -145,17 +145,33 @@ class BrowserPopupUtils {
+
+    let popoutWindowOptions: chrome.windows.CreateData;
+
+    if (waylandLike) {
```
```suggestion
    // The `senderWindow.left` and `senderWindow.top` are consistently reported as `0` by Chrome on Wayland, this is due to Wayland's design philosophy.
    // If waylandLike we explicitly reset the new popoutWindow's top and left, allowing the Wayland compositor to use its default placement logic for the window, ensuring it opens reliably and in a sensible location, rather than attempting potentially invalid coordinates.
    if (waylandLike) {
```

dereknance committed:
Merge branch 'main' into pm-24289-wayland-popout

dereknance committed:
avoid delete pattern, also check for linux

This commit also makes the original solution a bit more concise and adds
tests.

dereknance committed:
Merge branch 'main' into pm-24289-wayland-popout
